kawin/Thermodynamics.py

```diff
@@ -400,6 +462,87 @@ def _interdiffusivitySingle(self, x, T):
             return Dnkj
             
 
+    def getTracerDiffusivity(self, x, T, removeCache = True, phase = None):
+        '''
+        Gets tracer diffusivity for element el at specified x and T
+        Requires TDB database to have mobility or diffusivity parameters
+
+        Parameters
+        ----------
+        x : float, array or 2D array
+            Composition
+            Float or array for binary systems
+            Array or 2D array for multicomponent systems
+        T : float or array
+            Temperature
+            If array, must be same length as x
+                For multicomponent systems, must be same length as 0th axis
+        removeCache : boolean
+        phase : str
+
+        Returns
+        -------
+        tracer diffusivity - will return array if T is an array
+        '''
+        td = []
+
+        if hasattr(T, '__len__'):
+            for i in range(len(T)):
+                td.append(self._tracerDiffusivitySingle(x[i], T[i], removeCache, phase))
+            return np.array(td)
+        else:
+            return self._tracerDiffusivitySingle(x, T, removeCache, phase)
+
+    def _tracerDiffusivitySingle(self, x, T, removeCache = True, phase = None):
+        '''
+        Gets tracer diffusivity at unique composition and temperature
+
+        Parameters
+        ----------
+        x : float or array
+            Composition
+        T : float
+            Temperature
+        el : str
+            Element to calculate diffusivity
+        
+        Returns
+        -------
+        Tracer diffusivity as a float
+        '''
+        if phase is None:
+            phase = self.phases[0]
+
+        if not hasattr(x, '__len__'):
+            x = [x]
+
+        #Remove first element if x lists composition of all elements
+        if len(x) == len(self.elements) - 1:
+            x = x[1:]
+
+        cond = self._getConditions(x, T, 0)
+
+        if removeCache:
+            self._matrix_cs = None
+            self._parentEq, self._matrix_cs = local_equilibrium(self.db, self.elements, [phase], cond,
+                                                        self.models, self.phase_records,
+                                                        composition_sets=self._matrix_cs)
+
+        cs_matrix = [cs for cs in self._matrix_cs if cs.phase_record.phase_name == phase][0]
+
+        if self.mobCallables[phase] is None:
+            #NOTE: This is note tested yet
+            Dtrace = tracer_diffusivity_from_diff(cs_matrix, self.diffCallables[phase], diffusivity_correction=self.mobility_correction)
```

Comment:
Do you have a test for this?

Reply:
I originally never had a database to test this on. I made a copy of the NiCrAl database I used in most of the multicomponent thermodynamic tests and replaced the mobility parameters with diffusivity parameters. This gives a system I can test for numerical accuracy. Tests for calculating interdiffusivity and tracerdiffusivity from either mobility models or diffusivity models were added.